### PR TITLE
Domain mgmt RenewButton: rewrite to TS functional component

### PR DESCRIPTION
**Pull request by jsnajdr**
Rewrites the class component for purchase renewal to a TypeScript functional component. Part of #64069.

**How to test:**
Verify the various usages of the `RenewButton` across Domain Management UI still work as before. There are no behavior changes.

### CODE REVIEW

File: client/my-sites/domains/domain-management/security/index.jsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ class Security extends Component {
 					</p>
 					{ selectedSite.ID && ! purchase && <QuerySitePurchases siteId={ selectedSite.ID } /> }
 					<RenewButton
-						primary={ true }
+						primary
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/security/index.jsx:111]: 👍

**Review by tyxla — APPROVED** [review]
Nice and clean, works well for all "Renew One" buttons I've tested on 👍 Thanks! :shipit:

File: client/my-sites/domains/domain-management/edit/card/renew-button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	compact,
+	primary,
+	selectedSite,
+	subscriptionId,
+	redemptionProduct,
+	reactivate,
+	customLabel,
+	tracksProps,
+	purchase,
+	disabled,
```
Comment (daledupreez) [inline client/my-sites/domains/domain-management/edit/card/renew-button/index.tsx]: Nit: can we keep the props and type definition in alphabetical order? Or at least the same order? There are a _lot_ of props here!
```suggestion
	compact?: boolean;
	customLabel?: ReactChild;
	disabled?: boolean;
	primary?: boolean;
	purchase: Purchase | null;
	reactivate?: boolean;
	redemptionProduct?: ProductListItem;
	selectedSite: SiteData;
	subscriptionId: number;
	tracksProps?: Record< string, string >;
};

function RenewButton( {
	compact,
	customLabel,
	disabled,
	primary,
	purchase,
	reactivate,
	redemptionProduct,
	selectedSite,
	subscriptionId,
	tracksProps,
```

Comment (jsnajdr) [reply client/my-sites/domains/domain-management/edit/card/renew-button/index.tsx]: > can we keep the props and type definition in alphabetical order?

OK, I fixed this by dividing the props into two groups: the essential mandatory ones (`purchase`, `selectedSite`, `subscriptionId`) and optional ones, and sorting each group alphabetically.

**Pull request merged**